src/inabox/inabox-resources.js

```diff
@@ -183,6 +212,41 @@ export class InaboxResources {
     this.passObservable_.fire();
     this.firstPassDone_.resolve();
   }
+
+  /**
+   * @return {?IntersectionObserver}
+   * @private
+   */
+  initIntersectionObserver_() {
+    if (this.intersectionObserver_) {
+      return this.intersectionObserver_;
+    }
+    const {IntersectionObserver} = this.win;
+    if (!IntersectionObserver) {
+      return null;
+    }
+    return (this.intersectionObserver_ = new IntersectionObserver(
+      entries => entries.forEach(triggerViewportCallbackFromIntersection),
+      {threshold: 0.5}
+    ));
+  }
+
+  /**
+   * @param {!Resource} resource
+   * @private
+   */
+  maybeObserveIntersections_(resource) {
+    const {element, debugid} = resource;
+    if (!triggerViewportCallbackOn.includes(element.tagName)) {
+      return;
+    }
+    const observer = this.initIntersectionObserver_();
+    if (!observer) {
+      return;
+    }
+    observer.observe(element);
```

Comment:
There should be `unobserve` somewhere.

Reply:
Doing on `remove()`, let me know if I'm missing somewhere else.